### ISSUE TITLE
How to get the coordinates of obstacle area?

### ISSUE THREAD

**Issue by SleepTenMinuteAgain** [reiniscimurs/DRL-robot-navigation#154]
Hello, how can I get the coordinates of the obstacle area, such as the first area
if -3.8 > x > -6.2 and 6.2 > y > 3.8:
        goal_ok = False
I didn't understand what it meant by the obstruction area.
![image](https://github.com/reiniscimurs/DRL-robot-navigation/assets/116744137/fe3b418b-b9d5-4c96-814e-f22bd7893b16)
Is that the red area in the picture？

**Comment by reiniscimurs** [comment]
Yes that seems correct. We do not want goals to be located too close to the obstacles so we hardcode zones around them where a goal and robots starting position should not be located.

To get the coordinates you can either look up the positions of obstacles in the world file or simply estimate them even from this image (each square is 1x1 meter and the origin is the blue line)

**Comment by SleepTenMinuteAgain** [comment]
Thank you for your answer, it is very helpful to me!
I have another question for you, how to move the self-contained aggregates (such as the cube below) in gazebo? Or is it moving at some linear and angular velocity?
![image](https://github.com/user-attachments/assets/7f9cf1fd-2355-441c-81ee-7d7d33cb5ca2)

**Comment by reiniscimurs** [comment]
You mean to add it as a dynamic obstacle? Sorry, that is outside the scope of this repo and I do not know off the top of my head. I am sure there is some tutorial out there on that.

**Comment by SleepTenMinuteAgain** [comment]
Thank you very much for your patient answer❤❤❤!!!
I tried to modify the minimum collision distance, but there were some problems.
When I set the minimum collision distance to anything less than 0.35, the following picture will appear, collision occurs but does not terminate, how is this?
![image](https://github.com/user-attachments/assets/46e56ce7-d0ca-4e92-9ffc-5b2c9d217059)

**Comment by reiniscimurs** [comment]
Are you sure a collision is detected here? What you see is a laser state representation. But we only use velodyne lidar to detect collisions. Check the min value of the collision check to see if any collision is actually detected here.
